model/device.go

```diff
@@ -183,6 +185,26 @@ func (gn GroupName) String() string {
 	return string(gn)
 }
 
+func (gn GroupName) Validate() error {
+	if len(gn) > 1024 {
+		return errors.New(
+			"Group name can at most have 1024 characters",
+		)
+	} else if len(gn) == 0 {
+		return errors.New(
+			"Group name cannot be blank",
+		)
+	} else if ok, _ := regexp.MatchString(
+		"^[A-Za-z0-9_-]*$", string(gn),
+	); !ok {
+		return errors.New(
+			"Group name can only contain: upper/lowercase " +
+				"alphanum, -(dash), _(underscore)",
+		)
+	}
+	return nil
+}
```

Comment:
and a general mean nitpick: where are the units for model/Validate?

Reply:
I added a simple unit test that covers all scenarios.